tools/StripFFXIVClientStructs/StripFFXIVClientStructs/StripFFXIVClientStructs.cs

```diff
@@ -78,96 +39,578 @@ class StripFFXIVClientStructs
+            path = Path.GetFullPath(path);
+
+            string dest;
+            if (args.Length > 2)
+            {
+                dest = Path.GetFullPath(args[2]);
+                if (!Directory.Exists(dest))
+                {
+                    Directory.CreateDirectory(dest);
+                }
+            }
+            else
+            {
+                dest = path;
+            }
+
             foreach (var file in Directory.GetFiles(path, "*.cs", SearchOption.AllDirectories))
             {
-                var tree = CSharpSyntaxTree.ParseText(File.ReadAllText(file));
-                var root = tree.GetRoot();
-                var outTree = recurseNode(root);
-                File.WriteAllText(file, outTree.ToString());
+                // Relative path to the current file
+                var relFile = file.Replace(path, "");
+                if (DiscardFiles.Any((discard) => relFile.StartsWith(discard)))
+                {
+                    // If we're transforming in-place, delete files that we don't want to keep
+                    if (path == dest)
+                    {
+                        File.Delete(file);
+                    }
+                    continue;
+                }
+
+                try
+                {
+                    var tree = CSharpSyntaxTree.ParseText(File.ReadAllText(file));
+                    var root = tree.GetRoot();
+                    // SyntaxRewriter must be on a per-file basis
+                    var rewriter = new SyntaxRewriter(ns);
+                    // We know that this will always be CompilationUnitSyntax because that's what we get back from
+                    // `CSharpSyntaxTree.ParseText` for any given fully valid C# file
+                    CompilationUnitSyntax outTree = (CompilationUnitSyntax)rewriter.Visit(root);
+
+                    // Add any missing global-level using statements
+                    foreach (var usingValTemplate in GlobalUsings)
+                    {
+                        var usingVal = string.Format(usingValTemplate, ns);
+
+                        if (!rewriter.usings.Contains(usingVal))
+                        {
+                            outTree = outTree.AddUsings(new[] {
+                                SyntaxFactory.UsingDirective(SyntaxFactory.ParseName(usingVal))
+                            });
+                        }
+                    }
+
+                    var destFile = file.Replace(path, dest);
+                    Directory.CreateDirectory(Directory.GetParent(destFile).FullName);
+                    // Call `NormalizeWhitespace()`, otherwise Roslyn will omit whitespace between nodes randomly 🤷🏼‍
+                    // Also call `ToFullString()` because sometimes it'll use the wrong output. I never figured out why this is needed.
+                    // The documentation doesn't indicate why, but sometimes it randomly returned old node text instead of replaced/udpated text?
```

Comment:
```suggestion
                    // The documentation doesn't indicate why, but sometimes it randomly returned old node text instead of replaced/updated text?
```

This is wild.  Thanks for the comment.